### PR TITLE
Alerting: Added support Alertmanager data source for upstream Prometheus AM implementation.

### LINKED ISSUE #35736: Alerting: support vanilla alertmanager
**What would you like to be added**:
Currently the alertmanager supports only cortex alertmanager implementation. Could also support the vanilla prometheus alertmanager, though you'd only be able to manage silences and view notifications.

**Why is this needed**:

### PR DESCRIPTION
**Pull request by domasx2**
**What this PR does / why we need it**:

This PR allows you to use a Alert Manager datasource with upstream Prometheus Alert Manager implementation. Previously only Cortex (Mimir/GEM) implementation was supported. When creating/editing a Alert Manager datasource, you can now select which implementation it is pointing to. 

In case of Prometheus Alert Manager, contact points & notification policies will be read-only in the UI, as this implementation does not have an API endpoint to update configuration.

I'll add documentation with a separate PR.

Bonus: drive-by minor css fixes and "No results." message for the Alert Groups page

https://user-images.githubusercontent.com/847684/135261713-b837a93c-487b-47b4-aace-e614843626bb.mp4

**Which issue(s) this PR fixes**:

Fixes #35736

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by mckn** [comment]
Nice work on this! @domasx2 what is the status of the alerting project? Would be nice to throw some ideas around regarding adding new plugin types for extending alerting.

I'm thinking on things like:

- Instead of having a data source for e.g. this scenario. Change so you can add an external alert manager as a plugin.
- Maybe also make it possible to add contact point types.

Not sure if this fits into the alerting road map but it feels like a natural next step to make the new alerting feature a bit more extensible and powerful.

**Comment by domasx2** [comment]
> Nice work on this! @domasx2 what is the status of the alerting project? Would be nice to throw some ideas around regarding adding new plugin types for extending alerting.
> Not sure if this fits into the alerting road map but it feels like a natural next step to make the new alerting feature a bit more extensible and powerful.

I'm spending only part of my time on alerting now and not for long. It might be better to bring up any new ideas in alerting slack channel, or as github feature requests :) 

> * Instead of having a data source for e.g. this scenario. Change so you can add an external alert manager as a plugin.
Can you elaborate on what would be the use case?  I think for connecting to other types of external alert delivery systems, a more straightforward solution that is already often used is to set up a webhook contact point that would push the alert to this system.

> * Maybe also make it possible to add contact point types.
That makes sense to me!

**Comment by mckn** [comment]
> > Nice work on this! @domasx2 what is the status of the alerting project? Would be nice to throw some ideas around regarding adding new plugin types for extending alerting.
> > Not sure if this fits into the alerting road map but it feels like a natural next step to make the new alerting feature a bit more extensible and powerful.
> 
> I'm spending only part of my time on alerting now and not for long. It might be better to bring up any new ideas in alerting slack channel, or as github feature requests :)
> 
> > * Instead of having a data source for e.g. this scenario. Change so you can add an external alert manager as a plugin.
> 
> Can you elaborate on what would be the use case? I think for connecting to other types of external alert delivery systems, a more straightforward solution that is already often used is to set up a webhook contact point that would push the alert to this system.
> 
> > * Maybe also make it possible to add contact point types.
> 
> That makes sense to me!

Thanks for the quick reply 🙏🏻  Will see if I can post this in the proper channels :)

**Review by peterholmberg — APPROVED** [review]
tested and works well

**Review by grobinson-grafana — DISMISSED** [review]

File: pkg/services/ngalert/api/lotex_am.go [hunk 1]
```diff
@@ -35,14 +50,57 @@ func NewLotexAM(proxy *AlertingProxy, log log.Logger) *LotexAM {
...
+		}
+		return ErrResp(http.StatusInternalServerError, err, "Unable to load datasource meta data")
+	}
+
+	impl := ds.JsonData.Get("implementation").MustString(defaultImplementation)
+	implEndpoints, ok := endpoints[impl]
+	if !ok {
+		return ErrResp(http.StatusBadRequest, fmt.Errorf("unsupported Alert Manager implementation \"%s\"", impl), "")
+	}
+	endpointPath, ok := implEndpoints[endpoint]
```
Comment (papagian) [inline pkg/services/ngalert/api/lotex_am.go:78]: Shouldn't we forbid also POST requests for the upstream?

Comment (domasx2) [reply pkg/services/ngalert/api/lotex_am.go:78]: It doesn't have a get/set config endpoint at all, but other endpoints work the same as with cortex. You can still post silences or alerts

**Review by papagian — APPROVED** [review]
I have tested it both with a Cortex and an upstream Alertmanager and works great 🎉

**Pull request merged**